c_src/shithouse.c

```diff
@@ -0,0 +1,76 @@
+// vim: noet ts=4 sw=4
+#include <errno.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+
+#include "shithouse.h"
+
+static void *_load(const char *filepath, size_t *len) {
+	void *to_return = NULL;
+	struct stat sb = {0};
+	int fd = open(filepath, O_RDONLY);
+	if (fd < 0)
+		return NULL;
+
+    fstat(fd, &sb);
+	*len = sb.st_size;
+	to_return = mmap(NULL, *len, PROT_READ, MAP_SHARED, fd, 0);
+	close(fd);
+
+	return to_return;
+}
+
+static void _unload(void *mmaped_region, const size_t len) {
+	munmap(mmaped_region, len);
+}
+
+int api_get(lua_State* L) {
```

Comment:
can this turned into a macro so parameterized on route and method to make it easier to work with?